components/data-bridge/org.wso2.carbon.databridge.agent/src/main/java/org/wso2/carbon/databridge/agent/endpoint/DataEndpointGroup.java

```diff
@@ -319,11 +320,17 @@ public void run() {
                         dataEndpoint.deactivate();
                     }
                 } else {
-                    String[] urlElements = DataPublisherUtil.getProtocolHostPort(
-                            dataEndpoint.getDataEndpointConfiguration().getReceiverURL());
-                    if (!isServerExists(urlElements[1], Integer.parseInt(urlElements[2]))) {
-                        dataEndpoint.deactivate();
+                    try {
+                        String[] urlElements = DataPublisherUtil.getProtocolHostPort(
+                                dataEndpoint.getDataEndpointConfiguration().getReceiverURL());
+                        if (!isServerExists(urlElements[1], Integer.parseInt(urlElements[2]))) {
+                            dataEndpoint.deactivate();
+                        }
+                    } catch (DataEndpointConfigurationException exception) {
+                        log.warn("Data Endpoint with receiver URL:" + dataEndpoint.getDataEndpointConfiguration().getReceiverURL()
+                                + " could not be deactivated", exception);
                     }
+
```

Comment:
@janakiThevan 
Shall we remove the additional line break... ? No need to add line break unless there is a logical separation of code that needs to be highlighted..